sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClient.java

```diff
@@ -164,25 +165,58 @@ public void upload(InputStream data, long length, boolean overwrite) {
      * @param requestConditions {@link BlobRequestConditions}
      * @param timeout An optional timeout value beyond which a {@link RuntimeException} will be raised.
      * @param context Additional context that is passed through the Http pipeline during the service call.
+     * @deprecated
      */
+    @Deprecated
     public void uploadWithResponse(InputStream data, long length, ParallelTransferOptions parallelTransferOptions,
```

Comment:
Could it be possible to leave this API around and hook up taking `Context` in `BlobOutputStream`? I feel that would be the best path moving forwards as using `BlobOutputStream` is lacking functionality to support tracing.